Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -608,43 +633,87 @@ final class PurchasesOrchestrator {
         } catch let error as PromotionalOffer.SignedData.Error {
             let error = ErrorUtils.invalidPromotionalOfferError(error: error,
                                                                 message: error.localizedDescription)
-            self.trackPurchaseEventIfNeeded(storeKitVersion: .storeKit2, error: error.asPublicError)
+            self.trackPurchaseEventIfNeeded(startTime,
+                                            successful: false,
+                                            productId: sk2Product.id,
+                                            promotionalOfferId: promotionalOffer?.identifier,
+                                            winBackOfferApplied: winBackOfferApplied,
+                                            storeKitVersion: .storeKit2,
+                                            purchaseResult: nil,
+                                            error: error.asPublicError)
             throw error
         } catch {
             let purchasesError = ErrorUtils.purchasesError(withStoreKitError: error)
-            self.trackPurchaseEventIfNeeded(storeKitVersion: .storeKit2, error: purchasesError.asPublicError)
+            self.trackPurchaseEventIfNeeded(startTime,
```

Comment:
I'm wondering, with all these exceptions happening here, and now increasing the lines of code, is much less legible. Does it make sense to extract processing the error into a separate method?

Reply:
Yeah, I've thought about that (thanks to the linter). I'll try to rewrite it to make it clearer because I agree that the code is not very easy to follow anymore. Thanks!